### ISSUE TITLE
Failed to compile when cargo is used as a dependency?

### ISSUE THREAD

**Issue by Frederick888** [rust-lang/cargo#4479]
This issues happens only with `0.21.0`. For instance, `cargo install cargo-vendor` fails due to:

```
error[E0004]: non-exhaustive patterns: `&__Nonexhaustive {  }` not covered
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/cargo-0.21.0/src/cargo/util/errors.rs:69:15
   |
69 |         match &self.0 {
   |               ^^^^^^^ pattern `&__Nonexhaustive {  }` not covered

error: aborting due to previous error

error: failed to compile `cargo-vendor v0.1.12`, intermediate artifacts can be found at `/tmp/cargo-install.mIDoD77bIShH`
```

I tried cleaning up all the caches but the issue still persists. Then I tested it again in a Docker container and exactly the same result was reproduced.

The source codes directly from GitHub can be compiled though. I reckon it's either an issue related to the packaging or something stupid that I've done...

**Comment by alexcrichton** [comment]
This was an unfortunate interaction with semver and error-chain, you'll want to pin to 0.11.0-rc.2 for error-chain to get Cargo compiling

**Comment by Frederick888** [comment]
@alexcrichton Thanks for your quick response. But how that could be achieved? For instance, how should I install `cargo-vendor` now?

**Comment by steveklabnik** [comment]
@alexcrichton i'd be interested in hearing more about what was the matter with semver here

also, error-chain did release 0.11.0 for real

**Comment by alexcrichton** [comment]
I've published 0.21.1 to fix this.

@steveklabnik the semver constraint was `error-chain = "0.11.0-rc.2"` which I thought w/ prerelease versions wouldn't end up matching the `0.11.0` version, but apparently it did?

**Comment by steveklabnik** [comment]
Yup, that's a bug: https://github.com/steveklabnik/semver/issues/124
